src/SemanticSearch/Backend.jl

```diff
@@ -100,7 +113,62 @@ function Corpus(corpus_name::Union{String, Nothing}=nothing, embedder_model_path
 
     # don't initialize hnsw yet because it needs initial features
     hnsw = nothing
-    return Corpus(db, hnsw, embedder, max_seq_len, [], 1)
+
+    # if not in-memory only, save info about the corpus as a json
+    if !isnothing(corpus_name)
+        corpus_data = Dict(
+            "embedder_model_path" => embedder_model_path,
+            "max_seq_len" => max_seq_len
+        )
+        json_str = JSON.json(corpus_data)
+        corpus_data_path = "$(CURR_DIR)/files/$(corpus_name)_data.json"
+        # test if the hnsw file exists
+        if isfile(corpus_data_path)
+            # file exists ; delete it
+            rm(corpus_data_path) 
+        end
+        open(corpus_data_path, "w") do file
+            write(file, json_str)
+        end
+    end
+
+    return Corpus(corpus_name, db, hnsw, embedder, max_seq_len, [], 1)
+end
+
+"""
+    function load_corpus(corpus_name)
+
+Loads an already-initialized corpus from its associated "artifacts" (relational
+database, vector index, and informational json).
+
+Parameters
+----------
+corpus_name : str
+    the name of your EXISTING vector database
+"""
+function load_corpus(corpus_name::String)
+    try
+        # sqlite database
+        db_path = "$(CURR_DIR)/files/$(corpus_name).db"
+        db = SQLite.DB(db_path)
+
+        # vector index
+        hnsw_path = "$(CURR_DIR)/files/$(corpus_name).bin"
+        hnsw = open(deserialize, hnsw_path)
+
+        # get corpus info from associated json
+        corpus_data_path = "$(CURR_DIR)/files/$(corpus_name)_data.json"
+        corpus_data = open(corpus_data_path, "r") do file
+            JSON.parse(read(file, String))
+        end
+        embedder_model_path = corpus_data["embedder_model_path"]
+        embedder = Embedder(embedder_model_path)
+        max_seq_len = corpus_data["max_seq_len"]
+        
+        return Corpus(corpus_name, db, hnsw, embedder, max_seq_len, [], 1)
+    catch e
+        throw(ArgumentError("Loading failed; corpus name not found."))
```

Comment:
Please add the `msg` from the caught error to aid with future (potential) debugging. Something like:

```suggestion
        throw(ArgumentError("Loading failed; corpus name not found.\n" + e.msg))
```

https://docs.julialang.org/en/v1/base/base/#Core.ErrorException

Reply:
This `e.msg` might fail if the thing we catch isn't an inheritor of `Core.ErrorException`, but I'd be surprised if anything is throwing non error objects around

Reply:
I'm not sure about this, but I think not all errors have have a message field, e.g. `SystemError`. I'll just check if `e` has a `msg` property and add it to the error if it does.